test/e2e/components/vl-map.js

```diff
@@ -1,6 +1,70 @@
 const { VlElement } = require('vl-ui-core').Test;
+const { By } = require('vl-ui-core').Test.Setup;
+const VlMapBaseLayer = require('./vl-map-baselayer');
+const VlMapSearch = require('./vl-map-search');
 
-class VlMap extends VlElement {  
+class VlMap extends VlElement {
+	async _getMap() {
+		return this.shadowRoot;
+	}
+
+	async getBaseLayers() {
+		const childElements = await this.findElements(By.css(this.selector + ' *'));
+		const tagNames = await Promise.all(childElements.map(element => element.getTagName()));
+		const baseLayerElements = childElements.filter((element, index) => tagNames[index].indexOf('vl-map-baselayer') > -1);
+		return Promise.all(baseLayerElements.map(async element => await new VlMapBaseLayer(this.driver, element)));
+	}
+	
+	async isEscapeKeyDisabled() {
+		return this.hasAttribute("disable-escape-key");
+	}
+
+	async isRotationDisabled() {
+		return this.hasAttribute("disable-rotation");
+	}
+	
+	async isMouseWheelZoomDisabled() {
+		return this.hasAttribute("disable-mouse-wheel-zoom");
+	}
+
+	async getOverviewMap() {
+		const map = await this._getMap();
+		await this.driver.wait(async () => {
+            const overviewMaps = await map.findElements(By.css('.ol-overviewmap'));
+			return overviewMaps.length > 0;
+		});
+		return await map.findElement(By.css('.ol-overviewmap'));
+	}
+
+	async getActiveBaseLayerTitle() {
+        return this.driver.executeScript(`
+			return arguments[0].map.baseLayers.find((layer) => {
+				return layer.getVisible();
+			}).get('title')
+				`, this);
+    }
+
+	async toggleBaseLayer() {
+		const overviewMap = await this.getOverviewMap();
+		return overviewMap.click();
+	}
+
+    async hasSearch() {
+        const search = await this.driver.executeScript(`return arguments[0].shadowRoot.querySelector('vl-map-search')`, this);
+        return search != null;
+    }
+
+    async getSearch() {
+        const search = await this.driver.executeScript(`return arguments[0].shadowRoot.querySelector('vl-map-search')`, this);
+        return new VlMapSearch(this.driver, search);
+    }
+
+    async hasZoom(zoom) {
+        return this.driver.wait(async () => {
+            const currentZoom = await this.driver.executeScript(`return arguments[0].map.getView().getZoom()`, this);
+            return currentZoom === zoom;
+        }, 2000).then(() => true).catch(() => false);
+    }
```

Comment:
Niet beter een `getZoomLevel` i.p.v. een `hasZoom`. Ik denk dat 't meer vrijheid geeft aan de eindgebruiker.

Reply:
Dacht ik ook eerst, maar probleem zat hem erin dat ik niet weet wanneer het zoomen gedaan is. Misschien is er nog een andere manier om daar achter te komen? Of het wachten nog ergens te steken?

Reply:
Ik zou ergens verwachten dat het wachten in de zoom functie steekt. Maar natuurlijk ga je ook moeten wachten op bv. de auto-extent.
Ik vind het op zich niet fout zo.

Ik zie trouwens dat er geen zoom functie is. Misschien moet die ook nog toegevoegd worden, want ik denk dat sommige afnemers dat wel gaan willen doen? Bijvoorbeeld als ze implementeren dat bepaalde zaken pas vanaf een bepaald zoom level te zien zijn.

Reply:
zoomIn en zoomOut geïmplementeerd